os/os-mac.h

```diff
@@ -117,3 +123,58 @@ static inline bool os_cpu_has(cpu_features feature)
 }
 
 #endif
+
+/*
+ * discard_pages should not be run within Rosetta. Native arm64 or x86 only. 
+ */
+
+#define MMAP_CHUNK_SIZE		(1024 * 1024 * 1024)
+
+static inline int discard_pages(int fd, off_t offset, off_t size)
+{
+	caddr_t *addr;
+	uint64_t chunk_size = MMAP_CHUNK_SIZE;
+
+	if (fsync(fd) < 0) {
+                int __err = errno; 
+		log_err("%s: Cannot fsync file: %s \n", __func__, strerror(errno));
+                errno = __err;
+		return -1;
+	}	    
+    
+	/*
+	 * mmap the file in 1GB chunks and msync(MS_INVALIDATE).
```

Comment:
I think you can skip this comment - it literally describing the loop so it adds very little...